### PR TITLE
Write down V1 specs

### PR DESCRIPTION
**Pull request by xPi2**
Original task: https://app.dework.xyz/nation3/court?taskId=654c7edf-69d2-4d8d-b3b6-c297d772dc1c

This is an attempt to write down most of the specs for developing the V1 of the project.

These specs are not written in stone and could slightly change during the development but should serve as a rough guide to the goals of this version.

Please read carefully, and don't hesitate to ask any doubts or suggest any change or request for clarification.

### CODE REVIEW

File: docs/V1/AgreementsUI.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+Additionally we send and URI of the agreement's metadata.
+
+The system will generate the hash of the terms from a file the user provides.
+The system will generate the criteria tree from a list of (account, balance) nodes.
+The system will generate the URI of the agreement's metadata from the metadata file.
+
+**Steps:**
+
+1. The user uploads the terms file.
+2. The system hash the file, stores the hash, and shows it to the user in the form.
```
Comment (aahna-ashina) [inline docs/V1/AgreementsUI.md:31]:
```suggestion
2. The system hashes the file, stores the hash, and shows it to the user in the form.
```

File: docs/V1/AgreementsUI.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+To perform only one transaction, we will use signed approvals ([EIP-2612](https://eips.ethereum.org/EIPS/eip-2612))
+
+We will retrieve the criteria tree & proofs from the metadata of the agreement to join.
+
+**Steps:**
+
+1. The system checks if the user can join the agreement.
+2. The user triggers the join button.
+3. The system generates an EIP-712 permit for the funds required to join.
+4. The user signs the permit on their wallet.
```
Comment (aahna-ashina) [inline docs/V1/AgreementsUI.md:64]:
```suggestion
4. The user signs the permit with their private key.
```

File: docs/V1/AgreementsUI.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+To join an agreement, we need to prove membership to the criteria tree and provide the required collateral.
+
+To perform only one transaction, we will use signed approvals ([EIP-2612](https://eips.ethereum.org/EIPS/eip-2612))
+
+We will retrieve the criteria tree & proofs from the metadata of the agreement to join.
+
+**Steps:**
+
+1. The system checks if the user can join the agreement.
+2. The user triggers the join button.
```
Comment (aahna-ashina) [inline docs/V1/AgreementsUI.md:62]:
```suggestion
2. The user presses the join button.
```

File: docs/V1/AgreementsUI.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+6. The user approves the transaction.
+
+### Finalize agreement
+
+To finalize an agreement, all the parties need to signal their willingness to finalize with the current positions.
+
+**Steps:**
+
+1. The system checks the user can finalize the agreement.
+2. The user triggers the finalize button.
```
Comment (aahna-ashina) [inline docs/V1/AgreementsUI.md:75]:
```suggestion
2. The user presses the finalize button.
```

File: docs/V1/AgreementsUI.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+4. The user approves the transaction.
+
+### Dispute agreement
+
+To raise a dispute, any of the parties can call the contract to dispute an agreement.
+
+**Steps:**
+
+1. The system checks if the user can dispute the agreement.
+2. The user triggers the dispute button.
```
Comment (aahna-ashina) [inline docs/V1/AgreementsUI.md:86]:
```suggestion
2. The user presses the dispute button.
```

File: docs/V1/AgreementsUI.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+3. The system generates the dispute transaction.
+4. The user approves the transaction.
+
+### Withdraw from the agreement
+
+When the contract is finalized, the parties can withdraw their positions.
+
+**Steps:**
+1. The system checks if the user can withdraw from the agreement & shows withdrawal details.
+2. The user triggers the withdraw button.
```
Comment (aahna-ashina) [inline docs/V1/AgreementsUI.md:96]:
```suggestion
2. The user presses the withdraw button.
```

File: docs/V1/AgreementsUI.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+3. The system generates the withdrawal transaction.
+4. The user approves the transaction.
+
+### Verify terms
+
+The users can verify if a terms file matches with the terms of the agreement.
+
+**Steps:**
+1. The system open dialog to select the terms file.
```
Comment (aahna-ashina) [inline docs/V1/AgreementsUI.md:105]:
```suggestion
1. The system opens a dialog to select the terms file.
```

File: docs/V1/AgreementsUI.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+2. The user pick the file.
```
Comment (aahna-ashina) [inline docs/V1/AgreementsUI.md:106]:
```suggestion
2. The user picks the file.
```

File: docs/V1/AgreementsUI.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+3. The system hash the file and check the hash against the agreement's hash. Prompt the user with the result.
```
Comment (aahna-ashina) [inline docs/V1/AgreementsUI.md:107]:
```suggestion
3. The system hashes the file and checks the hash against the agreement's hash. Then prompts the user with the result.
```

File: docs/V1/JurorsUI.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+4. The system generates the message to sign.
+5. The user signs the message.
+6. The system stores the signed message.
+
+### Approve resolution proposal
+
+Jurors can approve proposed resolution submissions.
+
+**Steps:**
+1. The user triggers the approve.
```
Comment (aahna-ashina) [inline docs/V1/JurorsUI.md:40]:
```suggestion
1. The user triggers the approval.
```

File: docs/V1/JurorsUI.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+2. The system generates the message to sign.
+3. The user signs the message.
+4. The system stores the signed message.
+
+### Reject resolution proposal
+
+Jurors can approve proposed resolution submissions.
+
+**Steps:**
+1. The user triggers the reject.
```
Comment (aahna-ashina) [inline docs/V1/JurorsUI.md:50]:
```suggestion
1. The user triggers the rejection.
```

File: docs/V1/JurorsUI.md [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+2. The system generates the message to sign.
+3. The user signs the message.
+4. The system stores the signed message.
+
+### Submit resolution proposal
+
+Once a resolution proposal is signed by the required jurors any of them can submit the resolution by on-chain execution.
+
+**Steps:**
+1. The user submit the resolution.
```
Comment (aahna-ashina) [inline docs/V1/JurorsUI.md:60]:
```suggestion
1. The user submits the resolution.
```

File: docs/V1/JurorsUI.md [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+4. The system confirms that the resolution has been appealed.
+
+### Execute resolution
+
+Once a submitted resolution passes the review period it can be executed by anyone.
+
+**Steps:**
+1. The user triggers the execution.
+2. The system generates the execution transaction.
+3. The ser approves the transaction.
```
Comment (aahna-ashina) [inline docs/V1/JurorsUI.md:82]:
```suggestion
3. The user approves the transaction.
```

File: docs/V1/JurorsUI.md [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+4. The system confirms that the resolution has been executed.
+
+### Verify terms
+
+The users can verify if a terms file matches with the terms of the agreement.
+
+**Steps:**
+1. The system open dialog to select the terms file.
```
Comment (aahna-ashina) [inline docs/V1/JurorsUI.md:90]:
```suggestion
1. The system opens a dialog to select the terms file.
```

File: docs/V1/JurorsUI.md [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+2. The user pick the file.
```
Comment (aahna-ashina) [inline docs/V1/JurorsUI.md:91]:
```suggestion
2. The user picks the file.
```

File: docs/V1/JurorsUI.md [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+3. The system hash the file and check the hash against the agreement's hash. Prompt the user with the result.
```
Comment (aahna-ashina) [inline docs/V1/JurorsUI.md:92]:
```suggestion
3. The system hashes the file and checks the hash against the agreement's hash. Then prompts the user with the result.
```

File: docs/V1/JurorsUI.md [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+
+## Views
+
+### Resolution creation
+
+This view will show the form required to create a resolution.
+
+Must provide:
+
+- A set of buttons to insert and edit the agreement positions
```
Comment (aahna-ashina) [inline docs/V1/JurorsUI.md:102]:
```suggestion
- A set of buttons to insert and edit the agreement positions.
```

File: docs/V1/SPEC.md [hunk 18]
```diff
@@ -0,0 +1,204 @@
+# Nation3 Court Jurisdiction V1 <!-- omit in toc -->
```
Comment (aahna-ashina) [inline docs/V1/SPEC.md:1]: Is `docs/V1/SPEC.md` a duplicate of `docs/SPEC.md`? Do we need to keep two versions of the same document?

**Review by aahna-ashina — APPROVED** [review]

File: docs/V1/SPEC.md (hunk 18 above)
Comment (xPi2) [reply docs/V1/SPEC.md:1]: I'm thinking of having the general spec of the system independent from the version and then a more detailed spec on how each version is implemented.

**Review by aahna-ashina — APPROVED** [review]

**Pull request merged**